interface/product_registration/product_registration_controller.javascript.inc.php

```diff
@@ -0,0 +1,129 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+?>
+<script type="text/javascript">
+"use strict";
+function ProductRegistrationController(productRegistrationService, jQueryWithjQueryUi) {
+    var self = this;
+
+    var _formTitleText = '<?php echo xla("OpenEMR Product Registration"); ?>';
+    var _emailNeededText = '<?php echo xla("Please provide a valid email address"); ?>';
+    var _registrationSuccessTitleText = '<?php echo xla("Success"); ?>';
+    var _registrationSuccessText = '<?php echo xla("Your installation of OpenEMR has been registered"); ?>';
+    var _productRegistrationService = productRegistrationService;
+    var _jQuery = jQueryWithjQueryUi;
+
+    self.getProductRegistrationStatus = function(callback) {
+        _productRegistrationService.getProductStatus(function(err, data) {
+            if (err) { return callback(err, null); }
+
+            callback(null, data);
+        });
+    };
+
+    self.showProductRegistrationModal = function() {
+        _displayFormView();
+    };
+
+    var _displayFormView = function() {
+        _jQuery('.product-registration-modal').dialog({
+            resizable: false,
+            height: "auto",
+            width: 400,
+            draggable: false,
+            title: _formTitleText,
+            modal: true,
+            buttons: {
+                '<?php echo xla("Submit"); ?>':  _formSubmissionHandler,
+                '<?php echo xla("No Thanks"); ?>': _formCancellationHandler
+            }
+        });
+
+        // Wire up "enter key" handler in case user doesn't click the modal buttons manually
+        _jQuery('.product-registration-modal .email').on('keypress', function(event) {
+            if (event.which == 13) {
+                _formSubmissionHandler();
+                return false;
+            }
+        });
+    };
+
+    var _formSubmissionHandler = function() {
+        var email = _jQuery('.product-registration-modal .email').val() || '';
+
+        if (email === '' || email.indexOf('@') < 0) {
+            _jQuery('.product-registration-modal .message').text(_emailNeededText);
+        } else {
+            _jQuery('.product-registration-modal .message').text('');
+
+            _productRegistrationService.submitRegistration(email, function(err, data) {
+                if (err) { return _registrationFailedHandler(err); }
+
+                _registrationCreatedHandler(data);
+            });
+        }
+    };
+
+    // If we are on the about_page, show the registration data.
+    self.displayRegistrationInformationIfDivExists = function(data) {
+      var _registrationEmail = '<?php echo xla("Registered email:"); ?>';
+      var _registrationId = '<?php echo xla("Registered id:"); ?>';
```

Comment:
As a standard we place the colon outside the translated constant (to avoid have a bunch of duplicate constants with and without the colon:
``````
var _registrationEmail = '<?php echo xla("Registered email"); ?>:';
var _registrationId = '<?php echo xla("Registered id"); ?>:';